### PR TITLE
Mesh remapping done via the grid always instead of the mesh

### PR DESCRIPTION
**Pull request by v1kko**
…refgeo map)

### CODE REVIEW

**Comment by v1kko** [comment]
@v1kko please use the correct variable names for the variables in the update mesh routine

File: src/mesh_creation_module.f90 [hunk 1]
```diff
@@ -12,444 +12,256 @@ MODULE mesh_creation_module
...
+
+    ! Copy structure template from reference topograpgy (for fine grid)
+    refgeo_fine = region%refgeo_init
+
+    do j = 1, ny_fine
+    do i = x1, x2
+
+      ! add the deltas to the original high resolution grid
+      refgeo_fine%Hi_grid( i,j) = MAX( 0._dp, region%refgeo_init%Hi_grid( i,j) + dHi_grid_fine( i,j))
+      refgeo_fine%Hb_grid( i,j) = region%refgeo_init%Hb_grid( i,j) + dHi_grid_fine( i,j)
```
Comment (v1kko) [inline src/mesh_creation_module.f90:240]: @jbrnlz , as a final check is the `Hb = Hb + dH**i**` correct here?

Comment (v1kko) [reply src/mesh_creation_module.f90:240]: if it is correct then I would say this pull request is ready to be merged

Comment (jbrnlz) [reply src/mesh_creation_module.f90:240]: lol you right, damn those copy pastes

File: src/mesh_creation_module.f90 [hunk 2]
```diff
@@ -237,7 +237,7 @@ SUBROUTINE update_mesh( region)
 
       ! add the deltas to the original high resolution grid
       refgeo_fine%Hi_grid( i,j) = MAX( 0._dp, region%refgeo_init%Hi_grid( i,j) + dHi_grid_fine( i,j))
-      refgeo_fine%Hb_grid( i,j) = region%refgeo_init%Hb_grid( i,j) + dHi_grid_fine( i,j)
+      refgeo_fine%Hb_grid( i,j) = region%refgeo_init%Hb_grid( i,j) + dHb_grid_fine( i,j)
       refgeo_fine%Hs_grid( i,j) = surface_elevation( dHi_grid_fine( i,j), dHb_grid_fine( i,j), 0._dp)
```
[the remark this answers is not in the document]
Comment (v1kko) [reply src/mesh_creation_module.f90:241]: go ahead, let me know when everything is ready :)

[the remark this answers is not in the document]
Comment (jbrnlz) [reply src/mesh_creation_module.f90]: Done.

**Pull request merged**